src/rest.cpp

```diff
@@ -464,12 +475,37 @@ static bool rest_block(const std::any& context,
 
 static bool rest_block_extended(const std::any& context, HTTPRequest* req, const std::string& uri_part)
 {
-    return rest_block(context, req, uri_part, TxVerbosity::SHOW_DETAILS_AND_PREVOUT);
+    return rest_block(context, req, uri_part, TxVerbosity::SHOW_DETAILS_AND_PREVOUT, /*part_offset=*/std::nullopt, /*part_size=*/std::nullopt);
 }
 
 static bool rest_block_notxdetails(const std::any& context, HTTPRequest* req, const std::string& uri_part)
 {
-    return rest_block(context, req, uri_part, TxVerbosity::SHOW_TXID);
+    return rest_block(context, req, uri_part, TxVerbosity::SHOW_TXID, /*part_offset=*/std::nullopt, /*part_size=*/std::nullopt);
+}
+
+static bool rest_block_part(const std::any& context, HTTPRequest* req, const std::string& uri_part)
+{
+    std::optional<size_t> part_offset{};
+    std::optional<size_t> part_size{};
+    try {
+        const auto part_offset_str = req->GetQueryParameter("offset");
+        if (part_offset_str.has_value()) {
+            part_offset = ToIntegral<size_t>(*part_offset_str);
+            if (!part_offset.has_value()) {
+                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part offset is invalid: %s", *part_offset_str));
+            }
+        }
+        const auto part_size_str = req->GetQueryParameter("size");
+        if (part_size_str.has_value()) {
+            part_size = ToIntegral<size_t>(*part_size_str);
+            if (!part_size.has_value()) {
+                return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Block part size is invalid: %s", *part_size_str));
+            }
+        }
+    } catch (const std::runtime_error& e) {
```

Comment:
```suggestion
    } catch (const std::runtime_error& e) [[unlikely]] {
```

Reply:
The above suggestion fails to compile on my machine:
```
src/rest.cpp: In function ‘bool rest_block_part(const std::any&, HTTPRequest*, const std::string&)’:
src/rest.cpp:505:43: error: expected ‘{’ before ‘[’ token
  505 |     } catch (const std::runtime_error& e) [[unlikely]] {
      |                                           ^

$ gcc --version
gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
```

Reply:
Ahh I didn't test these suggestions before submitting. I wasn't sure if we could have that annotation after a catch block. Please ignore.